migrate/src/main/scala/migrate/utils/ScalafixService.scala

```diff
@@ -0,0 +1,126 @@
+import migrate.utils.ScalaExtensions.OptionalExtension
+import scalafix.interfaces.Scalafix
+import scalafix.interfaces.ScalafixEvaluation
+
+final case class ScalafixService(
+  scalafix: Scalafix,
+  sources: Seq[AbsolutePath],
+  compilerOptions: Seq[String],
+  classpath: Classpath,
+  targetRootSemantic: AbsolutePath,
+  toolClasspath: Classpath
+) {
+  import ScalafixService._
+  lazy val scalafixClassLoader: ClassLoader = scalafix.getClass().getClassLoader()
+
+  def inferExplicitAndImplicits(): Try[ScalafixEvaluation] = evaluate(addExplicitResultTypesAndImplicits)
+
+  def fixSyntaxForScala3(): Try[ScalafixEvaluation] =
+    evaluate(fixSyntaxRules)
+
+  def fixInPlace(eval: ScalafixEvaluation): Unit =
+    if (eval.isSuccessful) {
+      val filesEvaluated = eval.getFileEvaluations.toSeq
+      filesEvaluated.foreach { oneFile =>
+        val absPath = AbsolutePath.from(oneFile.getEvaluatedFile).get
+        if (oneFile.isSuccessful) {
+          oneFile.previewPatchesAsUnifiedDiff().asScala match {
+            case None => scribe.info(s"Nothing to fix in $absPath)")
+            case Some(_) => {
+              oneFile.applyPatches()
+              scribe.info(s"Syntax fixed for $absPath)")
+            }
+          }
+        } else {
+          val errorMsg = oneFile.getErrorMessage.asScala.getOrElse("Unknown Error")
+          scribe.info(s"Failed to run scalafix with ${fixSyntaxRules.mkString(", ")} on $absPath because $errorMsg")
+        }
+      }
+    } else {
+      val errorMsg = eval.getErrorMessage.asScala.getOrElse("Unknown Error")
+      scribe.info(s"Failed to run scalafix with ${fixSyntaxRules.mkString(", ")} because $errorMsg")
+
+    }
+
+  def configure(
+    newSources: Seq[AbsolutePath],
+    newCompilerOptions: Seq[String],
+    newClasspath: Classpath,
+    newTargetRootSemantic: AbsolutePath
+  ): ScalafixService =
+    this.copy(
+      sources = newSources,
+      compilerOptions = newCompilerOptions,
+      classpath = newClasspath,
+      targetRootSemantic = newTargetRootSemantic
+    )
+
+  private def evaluate(rules: Seq[String]): Try[ScalafixEvaluation] = Try {
+    val classpathWithTargetSemantic = classpath :+ targetRootSemantic
+    val args = scalafix
+      .newArguments()
+      .withRules(rules.asJava)
+      .withPaths(sources.map(_.toNio).asJava)
+      .withClasspath(classpathWithTargetSemantic.paths.map(_.toNio).asJava)
+      .withScalacOptions(compilerOptions.asJava)
+      .withToolClasspath(toolClasspath.toUrlClassLoader(scalafixClassLoader))
+    args.evaluate()
+  }
+
+}
+
+object ScalafixService {
+  val fixSyntaxRules: Seq[String] = Seq(
+    "ProcedureSyntax",
+    "fix.scala213.ConstructorProcedureSyntax",
+    "fix.scala213.ExplicitNullaryEtaExpansion",
+    "fix.scala213.ParensAroundLambda",
+    "fix.scala213.ExplicitNonNullaryApply",
+    "fix.scala213.Any2StringAdd"
+  )
+  val addExplicitResultTypesAndImplicits: Seq[String] = Seq("MigrationRule", "ExplicitImplicits")
```

Comment:
Now that we have two rules, it is better not to have one that is named `MigrationRule`. I suggest `InferTypes` instead.

Reply:
ok